Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -192,79 +205,79 @@ struct CLuaFunctionParserBase
         // primitive types
         if constexpr (std::is_same_v<T, std::string> || std::is_same_v<T, std::string_view>)
             return (iArgument == LUA_TSTRING || iArgument == LUA_TNUMBER);
-        if constexpr (std::is_same_v<T, bool>)
+        else if constexpr (std::is_same_v<T, bool>)
```

Comment:
There is no need to add an `else` here and below. The _if statement_ above always ends in a _return_.

Reply:
> There is no need to add an `else` here and below. The _if statement_ above always ends in a _return_.

I added `else`s due to the nature of `if constexpr`. Not putting else in `if constexpr` will not disregard the latter statements. In other function it ends up with a compile time error

Reply:
Good to know, I expected the compiler to be better.